DragaliaAPI/Features/Trade/TradeService.cs

```diff
@@ -0,0 +1,213 @@
+﻿using DragaliaAPI.Database.Entities;
+using DragaliaAPI.Database.Repositories;
+using DragaliaAPI.Features.Reward;
+using DragaliaAPI.Features.Shop;
+using DragaliaAPI.Models;
+using DragaliaAPI.Models.Generated;
+using DragaliaAPI.Services.Exceptions;
+using DragaliaAPI.Shared.Definitions.Enums;
+using DragaliaAPI.Shared.MasterAsset;
+using DragaliaAPI.Shared.MasterAsset.Models;
+using DragaliaAPI.Shared.MasterAsset.Models.Trade;
+using Microsoft.EntityFrameworkCore;
+
+namespace DragaliaAPI.Features.Trade;
+
+public class TradeService : ITradeService
+{
+    private readonly ITradeRepository tradeRepository;
+    private readonly IRewardService rewardService;
+    private readonly ILogger<TradeService> logger;
+    private readonly IUserDataRepository userDataRepository;
+    private readonly IInventoryRepository inventoryRepository;
+    private readonly IPaymentService paymentService;
+
+    public TradeService(
+        ITradeRepository tradeRepository,
+        IRewardService rewardService,
+        ILogger<TradeService> logger,
+        IUserDataRepository userDataRepository,
+        IInventoryRepository inventoryRepository,
+        IPaymentService paymentService
+    )
+    {
+        this.tradeRepository = tradeRepository;
+        this.rewardService = rewardService;
+        this.logger = logger;
+        this.userDataRepository = userDataRepository;
+        this.inventoryRepository = inventoryRepository;
+        this.paymentService = paymentService;
+    }
+
+    public IEnumerable<TreasureTradeList> GetCurrentTreasureTradeList()
+    {
+        DateTimeOffset current = DateTimeOffset.UtcNow;
+
+        return MasterAsset.TreasureTrade.Enumerable
+            .Where(x => x.CompleteDate == DateTimeOffset.UnixEpoch || x.CompleteDate > current)
```

Comment:
Isn't `x.CompleteData > current` the opposite of the kind of filtering you want?

Additionally, do we want to respect commence / complete dates when they were based around the game's actual lifetime which we are now beyond?